Makefile

```diff
@@ -380,7 +380,10 @@ istioctl-install:
 .PHONY: test localTestEnv test-bins
 
 # Run coverage tests
-test: pilot-test mixer-test security-test broker-test galley-test common-test
+test:
+	set -o pipefail; \
+	$(MAKE) pilot-test mixer-test security-test broker-test galley-test common-test \
+	|& tee >(go-junit-report > junit.xml)
```

Comment:
I can't help but wonder if dumping to junit.xml will eventually be a problem, either because it'll by default dirty the source tree (e.g., it could be stored under "${ISTIO_OUT}/junit.xml" ) or because there will now be multiple targets using the same name (which would mostly matter for parallel execution [or possibly serial if someone forgets to move/rename the file between running different tests).  Anyway, just food for thought (vs an expected change).

Reply:
Those are my concerns too. I started the PR as an experiment and as a follow up I let the caller supply a destination file in place of junit.xml to address the concerns. Please take another look.